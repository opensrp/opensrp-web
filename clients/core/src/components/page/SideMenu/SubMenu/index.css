.side-collapse-nav {
    cursor: pointer;
    padding: 4px 12px 2px 12px;
}

.nav-item-extend {
    width: 100%;
    padding-left: 49px;
}

.collapse-menu-title {
    padding-left: 12px;
}

.active-collapse-menu {
    color: #007bff !important;
}
.side-nav-active {
    background: #deeef8;
    border-right: 4px solid #209fd9;
}
