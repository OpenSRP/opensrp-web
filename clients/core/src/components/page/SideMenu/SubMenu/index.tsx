import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import React, { Fragment } from 'react';
import { RouteComponentProps, withRouter } from 'react-router';
import { NavLink } from 'react-router-dom';
import { Collapse, Nav, NavItem } from 'reactstrap';
import './index.css';

/** interface for a module i.e group of related pages */
export interface NavigationModule {
  /** Page navigation for the module, collapses to reveal the page links */
  parentNav: ModulePageLink;
  /** pageLinks to pages under this navigational module */
  childNavs: PageLink[];
}

/** interface for objs describing links to different pages */
export interface PageLink {
  /** the link's displayable text */
  label: string;
  /** URL to get redirected on click */
  url: string;
}

/** interface for the module page links, module represents collection of related page links */
export interface ModulePageLink {
  /** text label visible on menu item */
  label: string;
  /** fontAwesome icon displayed on menu item */
  icon: IconProp;
  /** url that module link will redirect to */
  url?: string;
}

/** props for this component */
export interface SubMenuProps {
  /** object describing navigation data for the module page link */
  parentNav: ModulePageLink;
  /** array of the pageNavigation links grouped under this navigation module */
  childNavs: PageLink[];
  /** string literal with the label value of currently collapsed module navigation link */
  collapsedModuleLabel: string | null;
  /** callback to parent component changes the collapsedModuleLabel
   *  entry in the parent component
   */
  setCollapsedModuleLabel?: SetCollapsedModuleLabel;
}

/** this components state */
export interface SubMenuState {
  /** whether the child navigation links should be collapsed or hidden */
  isActiveModule: boolean;
}

/** takes a string and updates the current collapsed-module-label in the parent component
 * @param {string} label - label of clicked navigation module page link
 */
type SetCollapsedModuleLabel = (label: string) => void;

/** Default props */
export const defaultSubMenuProps: SubMenuProps = {
  childNavs: [],
  collapsedModuleLabel: '',
  parentNav: {
    icon: ['far', 'user'],
    label: 'home',
  },
};

/** intersection of all types describing the props */
type subMenuPropsTypes = SubMenuProps & RouteComponentProps;

/** The SubMenu component */
export class SubMenu extends React.Component<subMenuPropsTypes, SubMenuState> {
  public static defaultProps = defaultSubMenuProps;

  public render() {
    const { childNavs, parentNav, collapsedModuleLabel, location } = this.props;
    const pathName = location.pathname;
    /** whether to collapse the child pages for this navigation module */
    const collapseMenu: boolean = collapsedModuleLabel === parentNav.label;
    /** whether the navigation module is currently active; judging from the history.location pathname
     * set to true if one of the module navigation's children has a url similar to pathname or
     * if the url of the parentNav is similar to the pathname
     */
    const isActiveModule: boolean =
      childNavs.map(nav => nav.url).indexOf(pathName!) > -1 || parentNav.url === pathName;

    const moduleLinkJsx = (
      <Fragment>
        <FontAwesomeIcon icon={parentNav.icon} size="lg" />
        <span className="collapse-menu-title"> {parentNav.label} </span>
      </Fragment>
    );

    const moduleLinkClassName = classNames('nav', 'nav-link', 'side-nav-item', {
      'active-collapse-menu': isActiveModule,
    });

    return (
      <div id={`sub-menu-${parentNav.label.replace(' ', '-')}`}>
        <Nav
          id={parentNav.label.replace(' ', '-')}
          className="side-collapse-nav"
          onClick={this.setModuleLabel}
        >
          <NavItem id="module-link">
            {parentNav.url ? (
              <NavLink to={parentNav.url} className={moduleLinkClassName}>
                {moduleLinkJsx}
              </NavLink>
            ) : (
              <span className={moduleLinkClassName}>{moduleLinkJsx}</span>
            )}
          </NavItem>
        </Nav>

        <Collapse isOpen={collapseMenu}>
          {childNavs.map((childNavObj, key) => {
            return (
              <Nav key={'child-nav-' + key}>
                <NavItem className="nav-item-extend">
                  <NavLink
                    to={childNavObj.url}
                    className="nav-link side-nav-item"
                    activeClassName="side-nav-active"
                  >
                    <span> {childNavObj.label} </span>
                  </NavLink>
                </NavItem>
              </Nav>
            );
          })}
        </Collapse>
      </div>
    );
  }

  /** updates parent component with the label of the currently collapsed navigation module */
  private setModuleLabel = () => {
    const { setCollapsedModuleLabel, collapsedModuleLabel, parentNav } = this.props;

    const isLabelSet = collapsedModuleLabel === parentNav.label;
    // if label is already set then this will behave as a toggle
    isLabelSet && setCollapsedModuleLabel !== undefined
      ? setCollapsedModuleLabel('')
      : setCollapsedModuleLabel!(this.props.parentNav.label);
  };
}

export default withRouter(SubMenu);
