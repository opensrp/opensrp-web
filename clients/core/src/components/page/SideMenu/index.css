.side-nav-item {
    color: #3a484f;
    font-family: Roboto;
    font-weight: 500;
    line-height: 20px;
    text-align: left;
}

.side-nav-active {
    color: #0192d4 !important;
    font-family: Roboto;
    font-weight: 500;
    text-align: left;
}

.side-menu-extend {
    margin: 0;
    padding: 16px 0 0 0;
}

.side-menu-container {
    margin-top: 67px;
}
