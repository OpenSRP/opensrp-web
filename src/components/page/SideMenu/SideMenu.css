.side-nav-item {
  color: #3A484F;
  font-family: Roboto;
  font-weight: 500;
  line-height: 20px;
  text-align: left;
}

.side-nav-active {
  color: #0192D4 !important;
  font-family: Roboto;
  font-weight: 500;
  text-align: left;
}