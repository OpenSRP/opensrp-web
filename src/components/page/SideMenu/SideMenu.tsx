import * as React from 'react';

class SideMenu extends React.Component<{}, {}> {
  constructor(props: {}) {
    super(props);
  }

  public render() {
    return <div>Side Menu</div>;
  }
}

export default SideMenu;
