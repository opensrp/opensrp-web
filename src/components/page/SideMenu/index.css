.side-nav-item {
  color: #3a484f;
  font-weight: 500;
  line-height: 20px;
  text-align: left;
}

.side-nav-active {
  background: rgba(1, 146, 212, 0.1);
  color: #0192d4 !important;
  font-weight: 500;
  text-align: left;
}

.side-menu-extend {
  margin: 0;
  padding: 16px 0 0 0;
}

.main-container .hidden-container {
  display: none !important;
}

.side-menu-extend ul.nav {
  background: initial;
  background-color: initial;
  max-width: initial;
  width: 100%;
}

.main-container .sidebar {
  width: 300px;
}

.sidebar .side-menu-extend a:hover {
  text-decoration: none;
  color: #0192d4;
}

.sidebar .side-menu-container {
  position: relative;
  width: 100%;
  height: 100%;
  flex: 1 1;
  border: 1px;
  border-right-style: solid;
  border-right-color: #dfe3e8;
  background: #f8fafe;
}

.sidebar .side-menu-container .row {
  margin: auto;
}

.side-menu-container .side-menu-extend ul.side-collapse-nav .collapse {
  padding-right: initial;
  padding-left: 33px;
}
