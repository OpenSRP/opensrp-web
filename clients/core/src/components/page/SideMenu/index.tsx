import * as React from 'react';
import { Col, Row } from 'reactstrap';
import './index.css';
import SubMenu, { NavigationModule } from './SubMenu';

/** interface for object with aggregated information
 * on all navigation modules
 */
export interface AggregatedNavigationModule {
  /** Whether to display this navigation module */
  enabled: boolean;
  /** The navigation Module- includes what the parent navigation is
   * and what the child navigations that are nested under the parent
   * navigation are
   */
  module: NavigationModule;
}

/** interface for SideMenu component */
export interface SideMenuProps {
  /** An array of all the aggregatedNavigationModule - will include
   * all navigationModules whether enabled or disabled
   */
  navigationModules: AggregatedNavigationModule[];
}

/** default props for the sideMenu component */
const defaultSideMenuProps: SideMenuProps = {
  navigationModules: [],
};

/** interface for the local state for this component
 */
interface SideMenuState {
  /** label pointing to active navigation module */
  collapsedModuleLabel: string | null;
}

/** default state for SideMenu */
const defaultSideMenuState: SideMenuState = {
  collapsedModuleLabel: null,
};

/** SideMenu component */
class SideMenu extends React.Component<SideMenuProps, SideMenuState> {
  public static defaultProps = defaultSideMenuProps;
  constructor(props: SideMenuProps) {
    super(props);
    this.state = defaultSideMenuState;
  }

  public render() {
    const collapsedModuleLabel = this.state.collapsedModuleLabel;

    return (
      <div className="side-menu-container">
        <Row>
          <Col className="side-menu-extend">
            {this.props.navigationModules.map(({ enabled, module }, index) => {
              const props = {
                ...module,
                collapsedModuleLabel,
                setCollapsedModuleLabel: this.setCollapsedModuleLabel,
              };
              return enabled && <SubMenu key={index} {...props} />;
            })}
          </Col>
        </Row>
      </div>
    );
  }

  /** updates collapsedModuleLabel to label of the new navigation module that should be collapsed */
  private setCollapsedModuleLabel = (label: string) => {
    this.setState({
      collapsedModuleLabel: label,
    });
  };
}

export default SideMenu;
