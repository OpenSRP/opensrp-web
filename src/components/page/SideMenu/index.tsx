import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Col, Row } from 'reactstrap';
import { ENABLE_REPORT_MODULE } from '../../../configs/env';
import {
  ENABLE_ANC_PAGE,
  ENABLE_CHILD_PAGE,
  ENABLE_CLIENT_PAGE,
  ENABLE_HOUSEHOLD_PAGE,
  ENABLE_LOCATION_PAGE,
  ENABLE_ROLE_PAGE,
  ENABLE_TEAM_PAGE,
  ENABLE_USER_PAGE,
} from '../../../configs/env';
import {
  ADMIN,
  ALL_CLIENTS,
  ANC,
  ANC_URL,
  CHILD,
  CHILD_URL,
  CLIENT_RECORDS,
  CLIENT_URL,
  HOUSEHOLD,
  HOUSEHOLD_URL,
  LOCATIONS,
  LOCATIONS_URL,
  REPORTS,
  REPORTS_URL,
  ROLE_URL,
  ROLES,
  TEAM_URL,
  TEAMS,
  USER_URL,
  USERS,
} from '../../../constants';
import SubMenu, { ModulePageLink, NavigationModule, PageLink, SubMenuProps } from './SubMenu';

import { IconProp } from '@fortawesome/fontawesome-svg-core';
import './index.css';

// Page links

export const CLIENT_PAGE_NAVIGATION: PageLink = {
  label: ALL_CLIENTS,
  url: CLIENT_URL,
};
export const HOUSEHOLD_PAGE_NAVIGATION: PageLink = {
  label: HOUSEHOLD,
  url: HOUSEHOLD_URL,
};
export const ANC_PAGE_NAVIGATION: PageLink = {
  label: ANC,
  url: ANC_URL,
};
export const CHILD_PAGE_NAVIGATION: PageLink = {
  label: CHILD,
  url: CHILD_URL,
};

export const USERS_PAGE_NAVIGATION: PageLink = {
  label: USERS,
  url: USER_URL,
};
export const ROLES_PAGE_NAVIGATION: PageLink = {
  label: ROLES,
  url: ROLE_URL,
};
export const TEAMS_PAGE_NAVIGATION: PageLink = {
  label: TEAMS,
  url: TEAM_URL,
};
export const LOCATIONS_PAGE_NAVIGATION: PageLink = {
  label: LOCATIONS,
  url: LOCATIONS_URL,
};

// icons
export const clientModuleNavIcon: IconProp = ['far', 'user'];
export const reportModuleNavIcon: IconProp = ['fas', 'chart-line'];
export const adminModuleNavIcon: IconProp = ['fas', 'cog'];

// Navigation links for navigation module. collapses when clicked to display the pageLinks

export const CLIENT_MODULE_PARENT_NAV: ModulePageLink = {
  icon: clientModuleNavIcon,
  label: CLIENT_RECORDS,
};
export const REPORT_MODULE_PARENT_NAV: ModulePageLink = {
  icon: reportModuleNavIcon,
  label: REPORTS,
  url: REPORTS_URL,
};
export const ADMIN_MODULE_PARENT_NAV: ModulePageLink = {
  icon: adminModuleNavIcon,
  label: ADMIN,
};

// nav module objects
export const CLIENT_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [
    ENABLE_CLIENT_PAGE && CLIENT_PAGE_NAVIGATION,
    ENABLE_HOUSEHOLD_PAGE && HOUSEHOLD_PAGE_NAVIGATION,
    ENABLE_ANC_PAGE && ANC_PAGE_NAVIGATION,
    ENABLE_CHILD_PAGE && CHILD_PAGE_NAVIGATION,
  ].filter((childNav): childNav is PageLink => typeof childNav !== 'boolean'),
  parentNav: CLIENT_MODULE_PARENT_NAV,
};
export const REPORT_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [],
  parentNav: REPORT_MODULE_PARENT_NAV,
};
export const ADMIN_NAVIGATION_MODULE: NavigationModule = {
  childNavs: [
    ENABLE_USER_PAGE && USERS_PAGE_NAVIGATION,
    ENABLE_ROLE_PAGE && ROLES_PAGE_NAVIGATION,
    ENABLE_TEAM_PAGE && TEAMS_PAGE_NAVIGATION,
    ENABLE_LOCATION_PAGE && LOCATIONS_PAGE_NAVIGATION,
  ].filter<PageLink>(
    (childNav: PageLink | boolean): childNav is PageLink => typeof childNav !== 'boolean'
  ),
  parentNav: ADMIN_MODULE_PARENT_NAV,
};

/** Enable Clients Module from its child pages */
export const ENABLE_CLIENT_RECORDS_MODULE =
  ENABLE_ANC_PAGE || ENABLE_CLIENT_PAGE || ENABLE_HOUSEHOLD_PAGE || ENABLE_HOUSEHOLD_PAGE;
export type ENABLE_CLIENT_RECORDS_MODULE = typeof ENABLE_CLIENT_RECORDS_MODULE;

/** Enable Admin Module from its child pages */
export const ENABLE_ADMIN_MODULE =
  ENABLE_USER_PAGE || ENABLE_ROLE_PAGE || ENABLE_TEAM_PAGE || ENABLE_LOCATION_PAGE;
export type ENABLE_ADMIN_MODULE = typeof ENABLE_ADMIN_MODULE;

/** interface for the local state for this component
 * @property {string} collapsedModuleLabel - label pointing to active navigation module
 */
export interface SideMenuState {
  collapsedModuleLabel: string;
}

const defaultSideMenuState: SideMenuState = {
  collapsedModuleLabel: '',
};

class SideMenu extends React.Component<{}, SideMenuState> {
  constructor(props: RouteComponentProps) {
    super(props);
    this.state = defaultSideMenuState;
  }

  public render() {
    const collapsedModuleLabel = this.state.collapsedModuleLabel;

    // props for subMenu components

    const clientSubMenuProps: SubMenuProps = {
      ...CLIENT_NAVIGATION_MODULE,
      collapsedModuleLabel,
      setCollapsedModuleLabel: this.setCollapsedModuleLabel,
    };
    const reportSubMenuProps: SubMenuProps = {
      ...REPORT_NAVIGATION_MODULE,
      collapsedModuleLabel,
      setCollapsedModuleLabel: this.setCollapsedModuleLabel,
    };
    const adminSubMenuProps: SubMenuProps = {
      ...ADMIN_NAVIGATION_MODULE,
      collapsedModuleLabel,
      setCollapsedModuleLabel: this.setCollapsedModuleLabel,
    };
    return (
      <div className="side-menu-container">
        <Row>
          <Col className="side-menu-extend">
            {ENABLE_CLIENT_RECORDS_MODULE && <SubMenu {...clientSubMenuProps} />}
            {ENABLE_REPORT_MODULE && <SubMenu {...reportSubMenuProps} />}
            {ENABLE_ADMIN_MODULE && <SubMenu {...adminSubMenuProps} />}
          </Col>
        </Row>
      </div>
    );
  }

  /** updates collapsedModuleLabel to label of the new navigation module that should be collapsed */
  private setCollapsedModuleLabel = (label: string) => {
    this.setState({
      collapsedModuleLabel: label,
    });
  };
}

export default SideMenu;
