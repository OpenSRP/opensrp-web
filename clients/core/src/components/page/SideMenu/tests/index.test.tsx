import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import SideMenu from '..';
import {
    CLIENT_NAVIGATION_MODULE,
    ADMIN_NAVIGATION_MODULE,
    REPORT_NAVIGATION_MODULE,
} from '../../../../configs/navigationConfigs';

const history = createBrowserHistory();

describe('components/page/SideMenu', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    const sideMenuProps = {
        navigationModules: [
            {
                enabled: true,
                module: CLIENT_NAVIGATION_MODULE,
            },
            {
                enabled: true,
                module: REPORT_NAVIGATION_MODULE,
            },
            {
                enabled: true,
                module: ADMIN_NAVIGATION_MODULE,
            },
        ],
    };

    it('renders without crashing', () => {
        shallow(
            <Router history={history}>
                <SideMenu {...sideMenuProps} />
            </Router>,
        );
    });

    it('renders side menu correctly', () => {
        const wrapper = mount(
            <Router history={history}>
                <SideMenu {...sideMenuProps} />
            </Router>,
        );
        expect(
            wrapper
                .find('.side-menu-container div')
                .at(0)
                .find('.row div')
                .at(0)
                .find('.side-menu-extend div')
                .find('#sub-menu-Client-Records').length,
        ).toEqual(1);
        expect(
            wrapper
                .find('.side-menu-container div')
                .at(0)
                .find('.row div')
                .at(0)
                .find('.side-menu-extend div')
                .find('#sub-menu-Reports').length,
        ).toEqual(1);
        expect(
            wrapper
                .find('.side-menu-container div')
                .at(0)
                .find('.row div')
                .at(0)
                .find('.side-menu-extend div')
                .find('#sub-menu-Admin').length,
        ).toEqual(1);
        wrapper.unmount();
    });

    it('does not display any nav items if no navigation modules are enabled ', () => {
        const sideMenuProps = {
            navigationModules: [
                {
                    enabled: false,
                    module: CLIENT_NAVIGATION_MODULE,
                },
                {
                    enabled: false,
                    module: REPORT_NAVIGATION_MODULE,
                },
                {
                    enabled: false,
                    module: ADMIN_NAVIGATION_MODULE,
                },
            ],
        };
        const wrapper = mount(
            <Router history={history}>
                <SideMenu {...sideMenuProps} />
            </Router>,
        );
        expect(
            wrapper
                .find('.side-menu-container div')
                .at(0)
                .find('.row div')
                .at(0)
                .find('.side-menu-extend div')
                .at(0)
                .find('#sub-menu-Client-Records').length,
        ).toEqual(0);
        expect(
            wrapper
                .find('.side-menu-container div')
                .at(0)
                .find('.row div')
                .at(0)
                .find('.side-menu-extend div')
                .at(0)
                .find('#sub-menu-Reports').length,
        ).toEqual(0);
        expect(
            wrapper
                .find('.side-menu-container div')
                .at(0)
                .find('.row div')
                .at(0)
                .find('.side-menu-extend div')
                .at(0)
                .find('#sub-menu-Admin').length,
        ).toEqual(0);
    });
});
