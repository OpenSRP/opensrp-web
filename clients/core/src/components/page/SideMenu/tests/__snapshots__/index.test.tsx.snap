// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/page/SideMenu renders side menu correctly 1`] = `
<ul
  className="side-collapse-nav nav"
  id="Client-Records"
  onClick={[Function]}
>
  <NavItem
    id="module-link"
    tag="li"
  >
    <li
      className="nav-item"
      id="module-link"
    >
      <span
        className="nav nav-link side-nav-item"
      >
        <FontAwesomeIcon
          icon={
            Array [
              "far",
              "user",
            ]
          }
          size="lg"
        >
          <i
            className="fa"
          />
        </FontAwesomeIcon>
        <span
          className="collapse-menu-title"
        >
           
          Client Records
           
        </span>
      </span>
    </li>
  </NavItem>
</ul>
`;

exports[`components/page/SideMenu sets the collapsedModuleLabel correctly from clicks on parentNavs: Everything 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -112,11 +112,11 @@
                        \\"label\\": \\"All clients\\",
                        \\"url\\": \\"/clients\\",
                      },
                    ]
                  }
-                 collapsedModuleLabel={null}
+                 collapsedModuleLabel=\\"Client Records\\"
                  key=\\"0\\"
                  parentNav={
                    Object {
                      \\"icon\\": Array [
                        \\"far\\",
@@ -134,11 +134,11 @@
                          \\"label\\": \\"All clients\\",
                          \\"url\\": \\"/clients\\",
                        },
                      ]
                    }
-                   collapsedModuleLabel={null}
+                   collapsedModuleLabel=\\"Client Records\\"
                    history={
                      Object {
                        \\"action\\": \\"POP\\",
                        \\"block\\": [Function block],
                        \\"createHref\\": [Function createHref],
@@ -238,11 +238,11 @@
                      <Collapse
                        appear={false}
                        enter={true}
                        exit={true}
                        in={false}
-                       isOpen={false}
+                       isOpen={true}
                        mountOnEnter={false}
                        onEnter={[Function noop]}
                        onEntered={[Function noop]}
                        onEntering={[Function noop]}
                        onExit={[Function noop]}
@@ -254,11 +254,11 @@
                      >
                        <Transition
                          appear={false}
                          enter={true}
                          exit={true}
-                         in={false}
+                         in={true}
                          mountOnEnter={false}
                          onEnter={[Function noop]}
                          onEntered={[Function bound onEntered]}
                          onEntering={[Function bound onEntering]}
                          onExit={[Function bound onExit]}
@@ -266,12 +266,16 @@
                          onExiting={[Function bound onExiting]}
                          timeout={350}
                          unmountOnExit={false}
                        >
                          <div
-                           className=\\"collapse\\"
-                           style={Object {}}
+                           className=\\"collapsing\\"
+                           style={
+                             Object {
+                               \\"height\\": 0,
+                             }
+                           }
                          >
                            <Nav
                              key=\\"child-nav-0\\"
                              tag=\\"ul\\"
                              vertical={false}"
`;
